src/org/etools/j1939_84/bus/j1939/J1939.java

```diff
@@ -186,6 +186,7 @@ public J1939(Bus bus, J1939DaRepository j1939DaRepository) {
      *            the pgn that's being requested
      * @return true if the message is an Acknowledgement/Nack for the given pgn
      */
+    /* FIXME why Integer and not int? */
```

Comment:
Int is used here because Generic Packet doesn’t have a PGN and will be used as the expected class that will be returned

Reply:
@battjt , I have this backwards. In some places we requesting a packet using the class only and not supplying the PGN.  Thus when the PGN isn’t supplied we are querying for the static PGN field on the expected class.  I think we need to spend sometime refactoring all this to pass in the PGN ‘sooner’ so lower level methods have the PGNs supplied.